game-app/game-core/src/main/java/games/strategy/triplea/ui/ProductionPanel.java

```diff
@@ -60,10 +60,13 @@ class ProductionPanel extends JPanel {
 
   protected JDialog dialog;
   private boolean bid;
+
+  private int lowestCost;
   final Action doneAction = SwingAction.of("Done", () -> dialog.setVisible(false));
 
   ProductionPanel(final UiContext uiContext) {
     this.uiContext = uiContext;
+    lowestCost = 100000;
```

Comment:
One thing I forgot to ask:  Where does this value of 100k come from? Is it just an arbitrarily high number or is there a deeper meaning?

Reply:
Arbitrary high number. If it was zero, then if the associated if statement was never triggered it would cause divide by zero.

Reply:
Ah okay, in this case is something speaking against using [`Integer.MAX_VALUE`](https://docs.oracle.com/en/java/javase/11/docs/api/java.base/java/lang/Integer.html#MAX_VALUE)? Also if you want to can assign the value in the same line where you define the variable. No need to put it into the constructor